src/Microsoft.HttpRepl.Tests/OpenApi/OpenApiV3EndpointMetadataReaderTests.cs

```diff
@@ -0,0 +1,215 @@
+        {
+            string json = @"{
+  ""openapi"": ""3.0.0"",
+  ""paths"": {
+    ""/pets"": {
+      ""post"": {
+        ""requestBody"": {
+          ""summary"": ""Create a pet"",
+          ""operationId"": ""createPets"",
+          ""responses"": {
+            ""201"": {
+              ""description"": ""Null response""
+            }
+          }
+        }
+      }
+    }
+  }
+}";
+            List<EndpointMetadata> endpointMetadata = GetEndpointMetadataList(json);
+
+            Assert.Single(endpointMetadata);
+            Assert.Equal("/pets", endpointMetadata[0].Path);
+            Assert.Empty(endpointMetadata[0].AvailableRequests);
+        }
+
+        [Fact]
+        public void Read_WithValidInput_ReturnsEndpointMetadata()
+        {
+            string json = @"{
+  ""openapi"": ""3.0.0"",
+  ""paths"": {
+    ""/pets"": {
+      ""get"": {
+        ""requestBody"": {
+          ""content"": {
+            ""summary"": ""List all pets"",
+            ""operationId"": ""listPets"",
+            ""parameters"": [
+              {
+                ""name"": ""limit"",
+                ""in"": ""query"",
+                ""required"": false,
+                ""schema"": {
+                  ""type"": ""integer"",
+                  ""format"": ""int32""
+                }
+              }
+            ],
+            ""responses"": {
+              ""200"": {
+                ""description"": ""An paged array of pets""
+              }
+            }
+          }
+        }
+      },
+      ""post"": {
+        ""requestBody"": {
+          ""content"": {
+            ""summary"": ""Create a pet"",
+            ""operationId"": ""createPets"",
+            ""responses"": {
+              ""201"": {
+                ""description"": ""Null response""
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}";
+            List<EndpointMetadata> endpointMetadata = GetEndpointMetadataList(json);
+            IReadOnlyDictionary<string, IReadOnlyDictionary<string, IReadOnlyList<Parameter>>> availableRequests = endpointMetadata[0].AvailableRequests;
+            KeyValuePair<string, IReadOnlyDictionary<string, IReadOnlyList<Parameter>>> firstRequest = availableRequests.First();
+            KeyValuePair<string, IReadOnlyDictionary<string, IReadOnlyList<Parameter>>> secondRequest = availableRequests.ElementAt(1);
+
+            Assert.Single(endpointMetadata);
+            Assert.Equal("/pets", endpointMetadata[0].Path);
+
+            Assert.Equal(2, availableRequests.Count);
+            Assert.Equal("get", firstRequest.Key);
+            Assert.Equal("post", secondRequest.Key);
+        }
+
+        [Fact]
+        public void CanHandle_WithNoOpenApiKeyInDocument_ReturnsFalse()
+        {
+            string json = @"{
+  ""info"": {
+    ""version"": ""v1""
+  },
+   ""paths"": {
+  }
+}";
+            Assert.False(CanHandle(json));
```

Comment:
Please keep tests in AAA form for better readability.

Reply:
Updated here:
https://github.com/aspnet/HttpRepl/commit/9ca88d9302e5c4830970960b32d34f99654c656b